File: இரண்டாம் சுற்று/36.Reverse String Without Disturbing Special Characters or Punctuations/main.c
/*
i/p:
A man, in the boat says : I see 1-2-3 in the sky

o/p:
y kse, ht ni3 21ee sIsy : a sta o-b-e ht nin amA
*/

#include<stdio.h>

char isChar(char s)
{
    if((s >= 'a' && s <= 'z') || (s >= 'A' && s <= 'Z') || (s >= '0' && s <= '9'))
    {
        return 1;
    }
    else
    {
        return 0;
    }
}

int main()
{
    char S[100];
    gets(S);
    int len = strlen(S);
    int i = 0, j = len -1;
    char temp;
    while(i < j)
    {
        if(isChar(S[i]) && isChar(S[j]))
        {
            temp = S[i];
            S[i] = S[j];
            S[j] = temp;
            i++;
            j--;
        }
        else
        {
            if(isChar(S[i]) == 0)
            {
                i++;
            }
            
            if(isChar(S[j]) == 0)
            {
                j--;
            }
        }
    }
    printf("%s",S);
}

File: இரண்டாம் சுற்று/36.Reverse String Without Disturbing Special Characters or Punctuations/main.java
/*
i/p:
A man, in the boat says : I see 1-2-3 in the sky

o/p:
y kse, ht ni3 21ee sIsy : a sta o-b-e ht nin amA
*/

import java.util.Scanner;

public class reverseStringNotPnctuatns {

    static boolean ischar(char s)
    {
        if((s>='a' && s<='z') || (s>='A' && s<='Z') || (s>='0' && s<='9'))
        {
            return true;
        }
        else
        {
            return false;
        }
    }
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        String  strng;
        strng = sc.nextLine();
        char[] str = strng.toCharArray();
        int left=0, right=str.length-1;
        char temp;

        while(left<right)
        {
            if(ischar(str[left]) && ischar(str[right]))
            {
                temp = str[left];
                str[left] = str[right];
                str[right] = temp;
                left++;
                right--;
            }
            else
            {
                if(ischar(str[left]) == false)
                {
                    left++;
                }
                if(ischar(str[right]) == false)
                {
                    right--;
                }
            }
        }
        for(int ind=0;ind<str.length;ind++)
        {
            System.out.print(str[ind]);
        }
    }
}
